db/src/types.rs

```diff
@@ -0,0 +1,251 @@
+// Copyright 2016 Mozilla
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+#![allow(dead_code)]
+
+use std::collections::{BTreeMap};
+
+/// Core types defining a Mentat knowledge base.
```

Comment:
Much as I love in-tree docs, this might do nicely as a sibling of [Querying](https://github.com/mozilla/mentat/wiki/Querying), with a pointer left here. More discoverable, more ability to add tables and images…